docs/publishing/publish-nanopublications.md

```diff
@@ -0,0 +1,220 @@
+    sub:sig npx:hasAlgorithm "RSA" ;
+        npx:hasPublicKey "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCmso7vmRO/Cp4Pt0RkJJkV5qfc1WFYU/jMtkdxxb5+lfIVXNV97XQnM1Tj4fkb/W6jkP6fHl8mj8Q7hl7VgUnQ6I+B7cMGpxW9Z8Br+JNx8DPMMt08VCH5+JMENPRKl91r7rF/YPWCAgL9eqXSixCNMNAj5RBmMTQoPuRkpgmt1wIDAQAB" ;
+        npx:hasSignature "aPZMJ3Md6X1PHYvXJiNoRUni9+1oS9faCfiPRRCrj4K/uZPN0J/znjxGuCUxoZRJ4b4RfSxmHFGRKfCFusJX+7Y3xuxYx4GYHzYhBciK7T5pO02V4w6sdwHLKd5E+Wcl0PTr2t3lEjq6yzY98wEXlZLAbaRDBJvzpg5xORifQDw=" ;
+        npx:hasSignatureTarget this: .
+
+    this: prov:generatedAtTime "2020-12-01T10:44:32.367084"^^xsd:dateTime ;
+        prov:wasAttributedTo <https://orcid.org/0000-0000-0000-0000> .
+}
+```
+
+## 📂 Publish from a file
+
+You can also easily sign and publish a Nanopublication from a file.
+
+```python
+from rdflib import ConjunctiveGraph
+from nanopub import Nanopub, NanopubConf, load_profile
+
+# 1. Create the config
+np_conf = NanopubConf(profile=load_profile(), use_test_server=True)
+
+# 2. Load the file in a RDFLib graph
+g = ConjunctiveGraph()
+g.parse("nanopub.trig")
+
+# 3. Make a Nanopublication object with this assertion
+np = Nanopub(conf=np_conf, rdf=g)
+
+# 4. Publish the Publication object.
+np.publish()
+print(np)
+```
+
+## 🖨️ Display more logs
+
+You can change the log level of your logger to display more logs from the nanopub library, which can be help when debugging.
+
+```python
+import rdflib
+from nanopub import Nanopub, NanopubConf, load_profile
+
+# Instantiate the logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler()
+formatter = logging.Formatter(
+    "%(asctime)s %(levelname)s: [%(module)s:%(funcName)s] %(message)s"
+)
+console_handler.setFormatter(formatter)
+logger.addHandler(console_handler)
+
+# Usual workflow to publish nanopubs
+np_conf = NanopubConf(
+    profile=load_profile(),
+    use_test_server=True,
+    add_prov_generated_time=True,
+    attribute_publication_to_profile=True,
+)
+
+my_assertion = rdflib.Graph()
+my_assertion.add((
+    rdflib.URIRef('www.example.org/timbernerslee'),
+    rdflib.RDF.type,
+    rdflib.FOAF.Person
+))
+
+np = Nanopub(
+    conf=np_conf,
+    assertion=my_assertion
+)
+np.publish()
+print(np)
+```
+
+## ⚙️ Configure the nanopubs
+
+The  `NanopubConf` class is used to create a reusable configuration for the nanopublications you create and publish. It enables you to define:
+
+* which triples will be automatically added to the `provenance` and `pubinfo` graphs
+* which user `Profile` to use
+* to which server nanopubs will be published
+
+```python
+import rdflib
+from nanopub import Nanopub, NanopubConf, load_profile
+
+np_conf = NanopubConf(
+    profile=load_profile(),
+
+    # Define which server must be used (will be production by default)
+    use_test_server=True,
+    use_server="http://test-server.nanopubs.lod.labs.vu.nl",
+
+    # Add at which date and time the nanopub was generated:
+    add_prov_generated_time=True,
```

Comment:
Nice - so we don't need to explicitly specify these anymore

Reply:
Yes, it's much easier to pass them to various functions. And we can quickly add more parameters if needed